### DISCUSSION TITLE
kernel help

### DISCUSSION THREAD

**Discussion by Gallasd** [idaholab/moose#21913] [Q&A General] [answered]
Hello, I want to carry out thermal topology optimization research based on moose. The optimized design variable is the thermal conductivity of each element.
Since my c++ level is very poor, I would like to ask if it is possible to give the thermal conductivity in the form of a vector in the .i file without modifying the source code. For example, a model with N elements is given a thermal conductivity vector of N × 1.

**Comment by hugary1995** [comment, marked as answer]
Yeah, you can use https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html together with https://mooseframework.inl.gov/source/userobject/PropertyReadFile.html
